controllers/admin/AdminTranslationsController.php

```diff
@@ -33,8 +36,8 @@ class AdminTranslationsControllerCore extends AdminController
     const DEFAULT_THEME_NAME = _PS_DEFAULT_THEME_NAME_;
     const TEXTAREA_SIZED = 70;
 
-    /** @var string : Link which list all pack of language */
-    protected $link_lang_pack = 'http://i18n.prestashop.com/translations/%ps_version%/available_languages.json';
+    /** @var string : URL for available language pack list */
+    protected $link_lang_pack;
```

Comment:
I believe in this controller this field is actually used to get the list of available languages from the API
It gets the content and parses it here:
https://github.com/PrestaShop/PrestaShop/blob/f45b27dc27581c766ae669df38bffd91d31a9ce1/controllers/admin/AdminTranslationsController.php#L213

But this is exactly what `RemoteLanguagePackLoader::getLanguagePackList()` does, so you could probably replace this internal management of file downloading and replace it with the appropriate service function This way you can even get rid of this `link_lang_pack` field since it won't be needed anymore

Reply:
In fact, it looks that `RemoteLanguagePackLoader::getLanguagePackList()` isn't used and the behavior isn't exactly the same between `Tools::file_get_contents` and native `file_get_contents` used by `getLanguagePackList()`.

I have to admit i didn't want to go deeper in this direction. Especially due to tests lack and troubles to mock `file_get_contents` (if i could mock easily http client, it would be easier).

Reply:
Thanks <3
Don't worry i'll enjoy to refactor everything I can but I think it is important to split changes across PR's

Reply:
Dixit the man who wanted to refactor the legacy parameters configuration on his first try 😅